mmdet/models/roi_heads/sparse_roi_head.py

```diff
@@ -139,6 +144,37 @@ def _bbox_forward(self, stage, x, rois, object_feats, img_metas):
 
         return bbox_results
 
+    def _mask_forward(self, stage, x, rois, attn_feats):
+        mask_roi_extractor = self.mask_roi_extractor[stage]
+        mask_head = self.mask_head[stage]
+        mask_feats = mask_roi_extractor(x[:mask_roi_extractor.num_inputs],
+                                        rois)
+        # do not support caffe_c4 model anymore
+        mask_pred = mask_head(mask_feats, attn_feats)
+
+        mask_results = dict(mask_pred=mask_pred)
+        return mask_results
+
+    def _mask_forward_train(self, stage, x, attn_feats, sampling_results, gt_masks, rcnn_train_cfg):
+
+        if sum([len(gt_mask) for gt_mask in gt_masks])==0:
```

Comment:
I suggest to remove this part which will leading unmatched computation graph in different gpu when use dist train. After remove this part, you also need to deal with 0-dim tensor in transformer.py. This is some known problem when I implement queryinst in my fork

Reply:
unmatched computation graph only happened when a batch with none gt-box or gt-mask, with mscoco dataset its ok

Reply:
This implementation would cause GPU stuck and we already updated it to tackle the problems. The latest implementation can handle none-gt cases correctly.

Reply:
Great